client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -83,116 +154,74 @@ export class SiteVerticalsSuggestionSearch extends Component {
 				verticalSlug: value,
 			}
 		);
+	};
 
-	onSiteTopicChange = ( value, isNavigating ) => {
-		const hasValue = !! value;
-		const valueLength = value.length || 0;
-		const valueLengthShouldTriggerSearch = valueLength >= this.props.charsToTriggerSearch;
-		const result = this.searchForVerticalMatches( value );
-
-		// TODO:
-		// Where to put the railcar code will be reconsidered.
-		if (
-			hasValue &&
-			valueLengthShouldTriggerSearch &&
-			// Don't trigger a search if there's already an exact, non-user-defined match from the API
-			! result
-		) {
+	/**
+	 * Callback to be passed to consuming component when the search field is updated.
+	 *
+	 * @param {String} value The new search value
+	 * @param {Bool} isSuggestionSelected Whether the user has selected a suggestion
+	 */
+	onSiteTopicChange = ( value, isSuggestionSelected = false ) => {
+		if ( value && value !== this.props.searchValue ) {
 			this.setState( { railcar: this.getNewRailcar() } );
 		}
 
-		this.updateVerticalData( result, value );
-
-		this.setState( {
-			isNavigating,
-		} );
+		// We debounce the update on the vertical while the user is typing
+		// to prevent unnecessary site preview updates
+		this.setState(
+			{
+				isSuggestionSelected,
+				inputValue: value,
+			},
+			() => this.updateVerticalDataDebounced( this.searchForVerticalMatches( value ), value )
```

Comment:
Since we'll debounce this anyway, could we move this as a separate call after `this.setState()`? When the call is debounced for a second, there's little difference if we wait or not for the `setState()` to finish its job.